wordpress-mobile/WordPress-Android
# Replace flow with live data in sources

I've been looking a bit into the flows/livedata and I think it doesn't make sense to build flows and call asLiveData. I think it also quite lowers the performance and doesn't improve readibility. I've changed the interface in this PR so that the `buildSource` method now returns live data. The only problem with this approach is that it stops us from running slower operations (flow runs as a suspend function by default). I've fixed this by adding coroutine scope as the parameter to the build source. I think the result is quite neat and more readable tbh. Let me know what you think! 

To test:
- Make sure you have the my site improvements flag turned on and the app is restarted
- Click around the my site to check if everything looks ok
- Switch between a few sites to see if the screen gets updated.

PR submission checklist:
